src/identify/index.js

```diff
@@ -118,8 +118,8 @@ function updateSelf (peerSelf, observedAddr) {
   }
 
   for (var i = 0; i < peerSelf.previousObservedAddrs.length; i++) {
-    if (peerSelf.previousObservedAddrs[i].toString() === omh.toString()) {
-      peerSelf.previousObserveredAddrs.splice(i, 1)
+    if (multiaddr(peerSelf.previousObservedAddrs[i]).toString() === omh.toString()) {
```

Comment:
Going to update that on master, so then that we can solve the ECONRESET issue in this PR with the tests you made